git_machete/tests/functional/test_machete.py

```diff
@@ -494,36 +495,20 @@ def test_branch_reappears_in_definition(self) -> None:
         expected_error_message: str = '.git/machete, line 5: branch `develop` re-appears in the tree definition. ' \
                                       'Edit the definition file manually with `git machete edit`'
 
-        with self.assertRaises(MacheteException) as e:
+        with pytest.raises(MacheteException) as e:
             self.launch_command('status')
         if e:
-            self.assertEqual(e.exception.parameter, expected_error_message,
-                             'Verify that expected error message has appeared a branch re-appears in tree definition.')
+            assert e.value.parameter == expected_error_message, 'Verify that expected error message has appeared a branch re-appears in tree definition.'
 
     @mock.patch('git_machete.utils.run_cmd', mock_run_cmd)  # to hide git outputs in tests
     def test_show(self) -> None:
         self.setup_discover_standard_tree()
 
-        self.assertEqual(
-            self.launch_command(
-                "show", "up",
-            ).strip(),
-            "hotfix/add-trigger"
-        )
+        assert self.launch_command("show", "up").strip() == "hotfix/add-trigger"
```

Comment:
Hmmm Python's `assert` is preferred to unittest's `assertEqual`? 😯 or does pytest override the behavior of `assert` in some way?

Reply:
**pytest** allows you to use the standard Python `assert` for verifying expectations and values in Python tests. No fancy `assertEqual` or `assertDictEqual` etc, just plain and simple `assert`.

Example of how **pytest** works with asserts can be found in [pytest documentation](https://docs.pytest.org/en/7.1.x/how-to/assert.html#assert)

For a detailed breakdown of how **pytest** handles expressions, see Benjamin Peterson's article [Behind the scenes of pytest's new assertion rewriting](http://pybites.blogspot.com/2011/07/behind-scenes-of-pytests-new-assertion.html).

Reply:
Oh! good to know, I switched from `stestr` recently since it did allow for storing test results in CI-compatible (junit-xml) format ☹️